src/v/security/acl.h

```diff
@@ -337,8 +386,11 @@ inline const acl_host acl_wildcard_host = acl_host::wildcard_host();
  * permitted to execute an operation on. When associated with a resource, it
  * describes if the principal can execute the operation on that resource.
  */
-class acl_entry {
+class acl_entry
+  : public serde::
+      envelope<acl_entry, serde::version<0>, serde::compat_version<0>> {
 public:
+    acl_entry() = default;
```

Comment:
It would be nice if we didn't have to create default constructors for lots of types.

Now we have a use case for this comment: https://github.com/redpanda-data/redpanda/pull/1078#discussion_r626497175

Reply:
i totally agree, for some of those types default constructors do no even  have sense.